### DISCUSSION TITLE
Update connection timed out

### DISCUSSION THREAD

**Discussion by kkim5181** [dbeaver/dbeaver#13139] [Q&A]
#### System information: 
- Operating system (distribution) and version :Windows 10 version 1809
- DBeaver version : 21.1.0
- Additional extensions

#### Connection specification:
- Database name and version
- Driver name
- Do you use tunnels or proxies (SSH, SOCKS, etc)? 

#### Describe the problem you're observing:
I checked Automatic Updates from Preferences setting and when I  click for "checking for updates", I get the following error.
![image](https://user-images.githubusercontent.com/86247993/122786154-01a36900-d282-11eb-81af-6705f8e3e349.png)
My company uses a proxy setting and is there any way to fix this error?

Thanks,

**Comment by Matvey16** [comment]
Have you tried editing proxy settings?
![2021-06-24_09h59_08](https://user-images.githubusercontent.com/82543000/123217110-e67b6800-d4d2-11eb-8cab-6b02823df946.png)

**Comment by kkim5181** [comment]
Thank you for the information. The issue is not resolved even though I added proxy host and port number for HTTP.  +
The error message is the same.
![image](https://user-images.githubusercontent.com/86247993/123310747-6bc34480-d4f4-11eb-8f83-98c93702bb46.png)

**Comment by p91paul** [comment]
You should also change Active Provider from direct (which means no proxy) to manual at the top of that preferences page.

**Comment by kkim5181** [comment]
Hello,

Thank you for the information.
I changed it to Manual and get the same error.

![image](https://user-images.githubusercontent.com/86247993/123427037-1e95b000-d592-11eb-91e9-423126a6251c.png)

![image](https://user-images.githubusercontent.com/86247993/123427154-3ec56f00-d592-11eb-8d9e-a0d15ae86805.png)
Do I set Host and Provider for Proxy bypass as well?

**Comment by Matvey16** [comment]
@kkim5181 Yes, please try setting this options. It can also be one of the things that cause this issue

**Comment by kkim5181** [comment]
@Matvey16 I added Host as a same host name and Manual in Provider in  Proxy Bypass but I still have the same issue.
Any idea?

**Comment by p91paul** [comment]
you probably need to set the same proxy for https as well, since I believe version check is done over https. Leave only socks empty.
Proxy bypass is useful only if you have some host you do NOT want to use the proxy for (i.e. an exception).

**Comment by kkim5181** [comment]
@p91paul I set the same proxy for https as well and get the following error.
![image](https://user-images.githubusercontent.com/86247993/124646473-af0da380-de62-11eb-9fed-79c88cbfde67.png)

I uncheck the auto update. 
Do you think it is because the auto update is set to disable?

**Comment by p91paul** [comment]
@kkim5181 nah, I think it's because proxy servers are a horrible technology that should die in a big fire :sob:

Yours is possibly trying to MITM https, breaking https connections while doing it.
You can still try setting proxy to "system" instead of manual, which should copy windows proxy settings.
Try also restarting dbeaver every time you change proxy settings.

**Comment by Matvey16** [comment]
@kkim5181 This seems to be pretty specific to a proxy that is used at your workplace. If editing https has not worked, I think you should contact your system administrator to try resolving that problem (he probably knows a lot more about specifics of your proxy than us)
